src/OrleansCodeGenerator/RoslynCodeGenerator.cs

```diff
@@ -72,22 +73,24 @@ public void AddGeneratedAssembly(string targetAssemblyName, GeneratedAssembly ge
         /// <summary>
         /// Generates code for all loaded assemblies and loads the output.
         /// </summary>
-        public void GenerateAndLoadForAllAssemblies()
+        public IReadOnlyList<GeneratedAssembly> GenerateAndLoadForAllAssemblies()
```

Comment:
I wonder if worth keep using those interfaces give the recent article we saw few days ago on main chat that mention the (big) overhead to iterate on collection interfaces assuming that we know that there is no other implementations that we will ever return here.